Setup/UpgradeSchema.php

```diff
@@ -73,6 +74,59 @@ public function upgrade(SchemaSetupInterface $setup, ModuleContextInterface $con
             ['bolt_parent_quote_id']
         );
 
+        $this->setupFeatureSwitchTable($setup);
+
         $setup->endSetup();
     }
+
+    private function setupFeatureSwitchTable(SchemaSetupInterface $setup) {
+        // If the table exists we do nothing. In the future, we can add migrations here based on
+        // current version from context, however for now just move on.
+        // The reason we do ugrade schema instead of install schema is that install schema is triggered *only*
+        // once during install. This makes debugging harder.
+        // However upgrade schema is triggered on every update and we get a chance to make changes as needed.
+        $tableCreated = $setup->getConnection()->isTableExists(FeatureSwitchConfig::TABLE_NAME);
+
+        if ($tableCreated) {
+            return;
+        }
+
+        // Create table now that we know it doesnt already exists.
+        $table = $setup->getConnection()
+          ->newTable($setup->getTable(FeatureSwitchConfig::TABLE_NAME))
+          ->addColumn(
+            'id',
+            \Magento\Framework\DB\Ddl\Table::TYPE_INTEGER,
+            null,
+            ['identity' => true, 'unsigned' => true, 'nullable' => false, 'primary' => true],
+            'ID'
+        )
+        ->addColumn(
+            FeatureSwitchConfig::SWITCH_NAME_COL,
```

Comment:
In general I'm against using variable reference in migrations - let's say someone in the future someone change the value of this constant. things will work fine on fresh install but update will be broken.

Reply:
thats fair, but if someone changes the name on our model in hail things will break too?

i'd rather use a const than use a string everywhere?